### ISSUE TITLE
Consider adding citus_remove_rebalance_strategy()

### ISSUE THREAD

**Issue by onurctirtir** [citusdata/citus#7283]
It'd be convenient to have citus_remove_rebalance_strategy() as opposite of citus_add_rebalance_strategy(). 

Then, we'd also want to expand the surface of pg_dist_rebalance_strategy_trigger_func() from `BEFORE OR INSERT` to `BEFORE OR INSERT OR DELETE` and perform necessary checks for DELETE as well.

**Comment by JelteF** [comment]
I agree a UDF would be nice here. But I don't think many checks would need to be performed for delete. The main thing I'd be worried about is ending up without a default rebalance strategy.
